giftcard/src/main/java/com/adyen/checkout/giftcard/GiftCardComponentCallback.kt

```diff
@@ -40,7 +42,7 @@ interface GiftCardComponentCallback : ComponentCallback<GiftCardComponentState>
      *
      * See https://docs.adyen.com/api-explorer/ for more information on the API documentation.
      *
-     * @param paymentMethodDetails The data from the partial payment method component.
+     * @param paymentComponentData The data from the partial payment method component.
      */
-    fun onBalanceCheck(paymentMethodDetails: PaymentMethodDetails) = Unit
+    fun onBalanceCheck(paymentComponentData: PaymentComponentData<GiftCardPaymentMethod>)
```

Comment:
I think we should also use `PaymentComponentState` everywhere for the sake of consistency